api/src/main/java/org/apache/iceberg/RewriteFiles.java

```diff
@@ -73,4 +73,12 @@ RewriteFiles rewriteFiles(Set<DataFile> dataFilesToReplace, Set<DeleteFile> dele
    * @return this for method chaining
    */
   RewriteFiles validateFromSnapshot(long snapshotId);
+
+  /**
+   * Set the sequenceNumber write in manifest-list file.
+   *
+   * @param sequenceNumber a sequenceNumber
+   * @return this for method chaining
+   */
+  RewriteFiles setSequenceNumber(long sequenceNumber);
```

Comment:
I think we need better documentation for what this method does and when you would call it, since this affects correctness.